File: 프로그래머스/lv2/92335. k진수에서 소수 개수 구하기/k진수에서 소수 개수 구하기.js
const isPrime = (n) => {
    for(let i = 2; i <= Math.sqrt(n); i++) {
        if (n % i === 0) return false;
    }
    return n > 1;
}
function solution(n, k) {
    return n
        .toString(k) // n 을 k 진수로 변환
        .split('0') // 0을 기준으로 스플릿
        .filter(Number) // '' 와 같은 빈 값 제거를 위해
        .map(Number) // 숫자로 변환
        .filter(isPrime) // 소수 판단
        .length // 개수 리턴;
}

File: 프로그래머스/lv2/92335. k진수에서 소수 개수 구하기/k진수에서 소수 개수 구하기.java
import java.util.*;
class Solution {
    int isPrime(String s) {
        if(s.equals("") || s.equals("1")) return 0; // 제외조건
        long n = Long.parseLong(s); //s가 int의 범위를 넘을 수 있음
        for(int i = 2; i < (int)Math.sqrt(n) + 1; i++) { // 소수 판별
            if(n % i == 0) return 0;
        }
        return 1;
    }
    public int solution(int n, int k) {
        String str = Integer.toString(n, k); // n을 k 진수로 변경
        int cnt = 0;
        for(String s : str.split("0")) cnt += isPrime(s); //0을 기준으로 분리하여 소수 판별
        return cnt;
    }
}